ui/src/storage/storage.tsx

```diff
@@ -66,17 +66,19 @@ export async function fetchUserData(
       continue;
     }
 
-    for (const group of cohort.groups) {
-      for (const criteria of group.criteria) {
-        const config = underlay.uiConfiguration.criteriaConfigs.find(
-          (config) => config.id === criteria.config.id
-        );
-        if (!config) {
-          throw new Error(
-            `Underlay has no support for criteria "${criteria.config.id}".`
+    for (const section of cohort.groupSections) {
+      for (const group of section.groups) {
+        for (const criteria of group.criteria) {
+          const config = underlay.uiConfiguration.criteriaConfigs.find(
+            (config) => config.id === criteria.config.id
           );
+          if (!config) {
+            throw new Error(
+              `Underlay has no support for criteria "${criteria.config.id}".`
```

Comment:
Kind of, though at least currently there's no information about the actual plugin stored, only the id of the criteria config. It's more flexible for backwards compatibility though does introduce the possibility of accidentally mismatching plugins. In any case, this code was only being used by the old local storage implementation. I'll remove it at some point and only updated because I had to.